be/src/runtime/local_tablets_channel.cpp

```diff
@@ -242,7 +242,7 @@ void LocalTabletsChannel::add_chunk(Chunk* chunk, const PTabletWriterAddChunkReq
     count_down_latch.wait();
 
     // We need wait all secondary replica commit before we close the channel
-    if (_is_replicated_storage && close_channel) {
+    if (_is_replicated_storage && close_channel && response->status().status_code() == TStatusCode::OK) {
```

Comment:
Why it will be hanged without this check?
As well, it's better to add a concise commit message in the pull request.